infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationConfigManager.java

```diff
@@ -41,98 +64,251 @@
+                                TextFormat.shortDebugString(session));
+                        generateLogicalGroupConfig(session);
+                        break;
+                    default: break;
+                }
+        });
+    }
+
+    private void generateLogicalGroupConfig(LogReplicationSession session) {
         Map<UUID, List<UUID>> streamToTagsMap = new HashMap<>();
+        Set<String> streamsToReplicate = new HashSet<>();
+        // Check if the local cluster is the Sink for this session. Sink side will honor whatever Source side send
+        // instead of relying on groupSinksMap to filter the streams to replicate.
+        boolean isSink = session.getSinkClusterId().equals(localClusterId);
+        Map<String, Set<String>> logicalGroupToStreams = new HashMap<>();
+        if (!isSink) {
+            groupSinksMap.entrySet().stream()
+                    .filter(entry -> entry.getValue().contains(session.getSinkClusterId()))
+                    .map(Map.Entry::getKey)
+                    .forEachOrdered(group -> {
+                        logicalGroupToStreams.put(group, new HashSet<>());
+                    });
+            log.debug("Targeted logical groups={}", logicalGroupToStreams.keySet());
+        }
 
         registryTableEntries.forEach(entry -> {
-            String tableName = getFullyQualifiedTableName(entry.getKey());
+            String tableName = TableRegistry.getFullyQualifiedTableName(entry.getKey());
+            UUID streamId = CorfuRuntime.getStreamID(tableName);
 
-            if (entry.getValue().getMetadata().getTableOptions().getIsFederated()) {
+            // Add merge only streams first
+            if (MERGE_ONLY_STREAMS.contains(streamId)) {
                 streamsToReplicate.add(tableName);
-
-                // Collect tags for this stream
-                UUID streamId = CorfuRuntime.getStreamID(tableName);
-                List<UUID> tags = streamToTagsMap.getOrDefault(streamId, new ArrayList<>());
-                tags.addAll(entry
-                    .getValue()
-                    .getMetadata()
-                    .getTableOptions()
-                    .getStreamTagList()
-                    .stream()
-                    .map(streamTag -> TableRegistry.getStreamIdForStreamTag(entry.getKey().getNamespace(), streamTag))
-                    .collect(Collectors.toList()));
-                streamToTagsMap.put(streamId, tags);
-            } else {
-                // Registry and ProtobufDescriptor tables do not have the is_federated flag but are to be replicated.
-                if (!tableName.equals(getFullyQualifiedTableName(CORFU_SYSTEM_NAMESPACE, REGISTRY_TABLE_NAME)) &&
-                    !tableName.equals(getFullyQualifiedTableName(CORFU_SYSTEM_NAMESPACE, PROTOBUF_DESCRIPTOR_TABLE_NAME))) {
-                    streamsToDrop.add(CorfuRuntime.getStreamID(tableName));
+                // Collect tags for merge-only stream
```

Comment:
I think the reason is pretty much the same as in FULL_TABLE model: some tables might be opened on Source side only, then it will cause some issue if LR writes the table's actual data while RegistryTable doesn't know it. I think for the drop table support done by George recently also kind of relied on this. (i.e. RegistryTable needs to know the stream)

Reply:
I'm not sure if I understand this fully. do we need to replicate registryTable info on the sink side as well?

Chris, may be we can add some more comments here?